libraries/Microsoft.Bot.StreamingExtensions/Payloads/HeaderSerializer.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using Microsoft.Bot.StreamingExtensions.Transport;
+
+namespace Microsoft.Bot.StreamingExtensions.Payloads
+{
+    internal static class HeaderSerializer
+    {
+        public const byte Delimiter = (byte)'.';
+        public const byte Terminator = (byte)'\n';
+        public const byte End = (byte)'1';
+        public const byte NotEnd = (byte)'0';
+
+        public const int TypeOffset = 0;
+        public const int TypeDelimiterOffset = 1;
+        public const int LengthOffset = 2;
+        public const int LengthLength = 6;
+        public const int LengthDelimeterOffset = 8;
+        public const int IdOffset = 9;
+        public const int IdLength = 36;
+        public const int IdDelimeterOffset = 45;
+        public const int EndOffset = 46;
+        public const int TerminatorOffset = 47;
+
+        public static int Serialize(Header header, byte[] buffer, int offset)
+        {
```

Comment:
This method is very odd. At the VERY LEAST please include a detailed comment around what it's doing and why. 

Creating a custom protocol encoding here seems very odd as encoding / encoding is always problematic. 

Why not re-use an existing one? Say URL Encode / Decode? Or any of a thousand other common, standard, ones.

Reply:
This encodes the very custom header for the streaming extensions. Encoding and decoding the header is key. The header is not "encoded" by other existing formatters.

Reply:
But agree, we can document this in a comment.